### PR TITLE
Implement HMM cost to filter samples

### LINKED ISSUE #179: Alternative/additional data filtering by HMM cost
A strategy worth testing for getting rid of some dodgy sequences and short-term time travellers is to exclude samples that exceed a given HMM cost (say, 10). If done naively, this would exclude all descendants of major VoCs, which is clearly a bad idea. However, if we only filter samples that form a cluster of size one under an attachment path this could work quite well. The idea is that if this really is a new VoC there will soon be multiple samples mapping to the same path, and then we'll add it to the ARG as before.

### PR DESCRIPTION
**Pull request by szhan**
Fix #179

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Some tests still failing for me in test_utils.py

File: sc2ts/inference.py [hunk 1]
```diff
@@ -222,6 +222,13 @@ def submission_date(self):
     def submission_delay(self):
         return (self.submission_date - self.date).days
 
+    def get_hmm_cost(self, num_mismatches):
+        # Note that Recombinants objects have total_cost.
+        # This bit of code is sort of repeated.
+        num_parents = len(np.unique([seg.parent for seg in self.path]))
```
Comment (jeromekelleher) [inline sc2ts/inference.py:228]: Why not ``len(self.path)  -1`` here? I don't see any reason to start thinking about unique parents at this point, it's just complexity.

**Review by jeromekelleher — COMMENTED** [review]
LGTM, one comment

File: sc2ts/inference.py (hunk 1 above)
Comment (szhan) [reply sc2ts/inference.py:228]: This is what I was wondering about in the discussion. I ended up just following `total_cost` in the `Recombinant` class. It looks like a function of the number of parents (`len(fwd.parents)`), but it is actually the number of copying nodes in the path.

**Comment by szhan** [comment]
> Some tests still failing for me in test_utils.py

Fixed.

File: sc2ts/inference.py (hunk 1 above)
Comment (szhan) [reply sc2ts/inference.py]: Updated.

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**